guardian/newsletters-nx
# Pb/s3 storage live

## What does this change?

- Adds an s3 implementation for persisting Launched newsletters
- Adds a set-up script to ensure that environment variables which we use to determine AWS credential usage are available
- Adds an additional policy to the ec2 instance with read/write permission for the s3 bucket
- Updates to some variable naming for added clarity
- Moves the bucket definition to the stacks (Not clear where it was before ?)
- Various UI tweaks to layout & correction of some typos

## How to test

The key change here is that the deployed application will now be writing draft and published newsletters to s3 so data can persist between deploys.

1. Deploy the branch to code
2. Run through the wizard data collect
3. Ensure launched newsletters appear in newsletters Newsletters view  and in the API response


The above should also work locally. You will need to run the new set-up script (see readme). 

1. Head to Janus and get some `devPlayground` credentials
2. Create a bucket in the console and add its name to the .env.local file created in the root of the newsletters api project
3. Use the new `npm run dev:s3` to start the app 

Try creating and launching a newsletter a detailed in the steps above to the deployed app
